internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go

```diff
@@ -229,8 +230,7 @@ func expandTKGVsphereNetworkSettings(data []interface{}) (network *tkgvspheremod
 	}
 
 	if v, ok := lookUpNetwork[apiServerPortKey]; ok {
-		apiServerPort := v.(int)
-		network.APIServerPort = int32(apiServerPort)
+		helper.SetPrimitiveValue(v, &network.APIServerPort, "network.APIServerPort")
```

Comment:
Is this the only instance to change? @shobha2626 since you have created this ticket can you check this change and see if you know any other resource which needs this similar change?